dsx-fw/Core/Src/main.c

```diff
@@ -74,7 +74,7 @@ int main(void)
   /* USER CODE BEGIN 1 */
 
 	// Initialize DSX data structure
-	DSX_data_t dsx_data;
+	volatile DSX_data_t dsx_data;
```

Comment:
Added volatile in front so every functions that take in a DSX_data_t must have volatile in it too.